draft-ietf-jsonpath-base.md

```diff
@@ -1467,44 +1467,30 @@ a type system is first introduced.
 
 Each parameter as well as the result of a function extension must have a declared type.
 
-A type is a set of instances.
 Declared types enable checking a JSONPath query for well-typedness
 independent of any argument the JSONPath query is applied to.
 
-{{tbl-types}} defines the available types in terms of abstract instances, where `v` denotes a value, and `nl` denotes
-a nodelist.
+{{tbl-types}} defines the available types in terms of the instances they contain.
 
-| Type                 | Abstract Instances                       |
-| :--                  | :----------------                        |
-| `ValueType`          | `Value(v)`, `Nothing`                    |
-| `LogicalType`        | `LogicalTrue`, `LogicalFalse`            |
-| `NodesType`       | `Nodes(nl)`                              |
+| Type                 | Instances                       |
+| :--                  | :------------------------------ |
+| `ValueType`          | JSON values or `Nothing`        |
+| `LogicalType`        | `LogicalTrue` or `LogicalFalse` |
+| `NodesType`          | Nodelists                       |
 {: #tbl-types title="Function extension type system"}
 
 Notes:
 
-* `ValueType` is an abstraction of a JSON value or `Nothing`.
-* `LogicalType` is an abstraction of the result of a `logical-expr`.
-  Its two instances, `LogicalTrue` and `LogicalFalse`, are not related to
-  the JSON literals `true` and `false` and have no direct syntactical representation in JSONPath.
-* `NodesType` is an abstraction of a `filter-query` (which appears
-  in a test expression or as a function argument).
-  Members of `NodesType` have no direct syntactical representation in JSONPath.
-
-The abstract instances above can be obtained from the concrete representations in {{tbl-typerep}}.
-
-| Abstract Instance | Concrete Representations                                           |
-| :---------------: | :----------------------:                                           |
-| `Value(v)`        | JSON value `v`                                                     |
-| `Nothing`         | A representation of the absence of a JSON value, distinct from the JSON literal `null`, e.g., from a Singular Query or `filter-query` resulting in an empty nodelist   |
-| `Nodes(nl)`       | A list of zero or more nodes, e.g., from a `filter-query` resulting in the nodelist `nl`, which may or may not be empty  |
-{: #tbl-typerep title="Concrete representations of abstract instances"}
+* The only instances that can be directly represented in JSONPath syntax are the literal
+  primitive values of `ValueType`.
```

Comment:
"literal _and_ primitive values"?

`true`, `false`, and `null` are commonly known as literals, and strings and numbers are primitive.  I don't think I've heard either set described by both terms.

Reply:
JavaScript - where JSON took its data types from - knows [primitive types](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#primitive_values) and [literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#literals). So "literal and primitive values" seems not to be wrong.

Reply:
Fixed.